### ISSUE TITLE
Unnecessary dialogs on Trezor when creating a new hidden wallet

### ISSUE THREAD

**Issue by andrewkozlik** [trezor/trezor-suite#7434]
When creating a new hidden wallet with the passphrase being entered in Suite, the flow is as follows:
1. Suite: "Enter passphrase". User enters passphrase in Suite.
2. Trezor: "Access hidden wallet? Next screen will show the passphrase!"
3. Trezor: "Use this passphrase?"
4. Suite: "Confirm empty hidden wallet". User enters passphrase in Suite.
5. Trezor: Access hidden wallet? Next screen will show the passphrase!
6. Trezor: Use this passphrase? ...

I believe that steps 5 and 6 are unnecessary. Suite should simply compare the passphrase that was entered in step 4 with the one that was entered in step 1, it doesn't need Trezor to do that. For this to work Suite will need to keep the passphrase or some fingerprint of the passphrase in memory until step 4 is reached.

To be clear, if the user is entering the passphrase on Trezor, then the current flow is correct, i.e. they should enter the passphrase twice on Trezor.

Rationale:
- Assume that malware on the user's computer is attempting a ransom attack by sending a different passphrase to Trezor than what the user actually entered. Also assume that the user doesn't notice the incorrect passphrase in step 3. Any malware that is able to tamper with the Suite-Trezor communication can also presumably force steps 5 and 6 to be skipped and can provide the expected responses to Suite as if they happened.
- Assume that the user just mistyped the passphrase for an existing wallet and didn't notice the typo in step 3. Then showing "Confirm empty hidden wallet" should be sufficient to alert them about this. Furthermore, forcing them to carefully retype the passphrase in step 4 should uncover this with near certainty. Making them confirm the passphrase for a fourth time is unlikely to add any value.
- Assume that the user is creating a new hidden wallet. In this case they are already forced to enter the passphrase twice and check the passphrase once on Trezor. That's already three confirmations of the new passphrase. Making them confirm it for a fourth time is unlikely to add any value.

**Comment by sime** [comment]
> I believe that steps 5 and 6 are unnecessary. Suite should simply compare the passphrase that was entered in step 4 with the one that was entered in step 1, it doesn't need Trezor to do that. For this to work Suite will need to keep the passphrase or some fingerprint of the passphrase in memory until step 4 is reached.

I assume the original intention was to replicate the experience of entering a duplicate password with a typical web service account creation.

Due to the fragile nature of passphrases the fourth confirmation step does enforce the importance of accurate entry.

Currently, I see this as an optimisation of an unhappy path.

**Comment by jvaclavik** [comment]
@Hermez-cz is it still relevant?

**Comment by Hermez-cz** [comment]
@jvaclavik Yes it is, it might simplify the confirmation passphrase (re)entry even in the new Suite Flow.

**Comment by vojtatranta** [comment]
This would be about eliminating the confirmation screens on Trezor and validation of the passphrase on the device itself. We won't do that. It may present a security risk.

**Comment by andrewkozlik** [comment]
> This would be about eliminating the confirmation screens on Trezor and validation of the passphrase on the device itself.

That is completely false. Confirmation on Trezor happens in steps 2 and 3.

**Comment by vojtatranta** [comment]
> > This would be about eliminating the confirmation screens on Trezor and validation of the passphrase on the device itself.
> 
> That is completely false. Confirmation on Trezor happens in steps 2 and 3.

Yes, of course. The confirmation on the device happens twice. This would eliminate the confirmation of the passphrase on the device in the confirmation step (4,5,6).
We do not store the raw passphrase anywhere, we only compare the returned hashes.

**Comment by andrewkozlik** [comment]
FTR, from internal discussion, the reason why this issue was closed was that the devs judged that the UX improvement is not worth the effort, because it was assessed as a significant change with a potential to break something and because the passphrase hash needs to be reliably removed from redux once it is no longer needed.
